api/tests/mocha/services/report/smsparser.spec.js

```diff
@@ -613,13 +613,49 @@ describe('sms parser', () => {
     chai.expect(data).to.deep.equal({patient_id: '12345', lmp_date: null});
   });
 
+  it('parse patient id in nepali correctly', () => {
+    const doc = {message:'०३१७७'};
+    const def = definitions.forms.YYYT;
+    const data = smsparser.parse(def, doc);
+    chai.expect(data).to.deep.equal({patient_id: '03177'});
+  });
+
+  it('parse patient id in nepali and english correctly', () => {
+    const doc = {message:'०३१77'};
+    const def = definitions.forms.YYYT;
+    const data = smsparser.parse(def, doc);
+    chai.expect(data).to.deep.equal({patient_id: '03177'});
+  });
+
+  it('parse lmp date in nepali correclty', () => {
+    const doc = {message:'०३१77 २०८० १ १'};
+    const def = definitions.forms.YYYS;
+    const data = smsparser.parse(def, doc);
+    chai.expect(data).to.deep.equal({
+      patient_id: '03177',
+      lmp_year: '2080', lmp_month: '1', lmp_day: '1',
+      lmp_date: moment('2023-04-14').valueOf()
+    });
+  });
+
+  it('parse lmp date in nepali and english correclty', () => {
+    const doc = {message:'०३१77 २०70 08 १4'};
+    const def = definitions.forms.YYYS;
+    const data = smsparser.parse(def, doc);
+    chai.expect(data).to.deep.equal({
+      patient_id: '03177',
+      lmp_year: '2070', lmp_month: '08', lmp_day: '14',
+      lmp_date: moment('2013-11-29').valueOf()
+    });
+  });
+
   it('parse BS date parts yyys 2: textforms', () => {
     const doc = { message: '#ID 12345 #Y 2068 #M 11 #D 29' };
     const def = definitions.forms.YYYS;
     const data = smsparser.parse(def, doc);
     chai.expect(data).to.deep.equal({
-      patient_id: 12345,
-      lmp_year: 2068, lmp_month: 11, lmp_day: 29,
+      patient_id: '12345',
+      lmp_year: '2068', lmp_month: '11', lmp_day: '29',
```

Comment:
This change is concerning to me - is there any way that the standardiseDigits function could be called with a number in production code, or was this an inaccuracy in the test code?

Reply:
I checked how these fields are stored ultimately with current change and before, and they're all stored in quotation in production in couchdb, so I guess this was inaccuracy in test code.